internal/getter/file.go

```diff
@@ -1,15 +1,48 @@
 package getter
 
 import (
+	"fmt"
 	"path/filepath"
+	"runtime"
 
 	"github.com/hashicorp/go-getter"
 	"github.com/spf13/afero"
 )
 
 // fileDetector is a replacement for go-getter's own file detector which also verifies the source file exists
 type fileDetector struct {
-	fd getter.FileDetector
+	fd               getter.FileDetector
+	allowNonExisting bool
+}
+
+func isValidPath(fs afero.Fs, path string) bool {
+	err := fs.MkdirAll(path, 0700)
+	if err == nil {
+		fs.Remove(path)
```

Comment:
This deletes only the top level directory. We could recurse and delete all empty ones going up, but I'm not sure I'm happy doing that